gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConfigureEndpointMinimumVersionsTask.java

```diff
@@ -63,8 +62,11 @@ final void action() {
         return endpointVersions;
     }
 
+    // TODO(fwindheuser): Replace 'JavaPluginConvention'  with 'JavaPluginExtension' after dropping Gradle 6 support.
+    @SuppressWarnings("deprecation")
     private Manifest createManifest(Project project, List<EndpointMinimumVersion> versions) {
-        JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
+        org.gradle.api.plugins.JavaPluginConvention javaConvention =
+                project.getConvention().getPlugin(org.gradle.api.plugins.JavaPluginConvention.class);
         return javaConvention.manifest(manifest -> {
```

Comment:
This is a bit unfortunate. The `JavaPluginConvention` class is deprecated and to be deleted in Gradle 8. The replacement class `JavaPluginExtension` exists since Gradle 4, but all of the methods, e.g. `JavaPluginExtension#manifest`, or `JavaPluginExtension#getSourceSets` were only added in Gradle 7.1.

I think its best to keep the deprecated methods for now. Once we migrate to Gradle 8 and depending on usage across the fleet, we can either cut support for Gradle 6 or add a workaround using reflection.